File: cos/cos.olympus/src/main/java/cos/olympus/game/Player.java
package cos.olympus.game;

final class Player implements Avatar {
    final int    id;
    final String name;
    int lastX;
    int lastY;

    public Player(int id, String name) {
        this.id = id;
        this.name = name;
    }

    public Player(int id, String name, int lastX, int lastY) {
        this.id = id;
        this.name = name;
        this.lastX = lastX;
        this.lastY = lastY;
    }

    @Override public int id() {
        return id;
    }

    @Override public String name() {
        return name;
    }

    @Override public int x() {
        return lastX;
    }

    @Override public int y() {
        return lastY;
    }
}

File: cos/cos.olympus/src/main/kotlin/cos/olympus/game/Player.kt
package cos.olympus.game

import cos.olympus.game.actions.Action
import cos.olympus.game.spells.SpellStrategy
import java.util.*

data class Player(
    override val id: Int,
    override val state: CreatureState
) : Creature {
    val zone = HashMap<Int, Creature>()
    val spellZone = HashMap<Long, SpellStrategy>()

    override val actions = ArrayDeque<Action>()
    override val name get() = "Player_$id"
    override val x get() = state.x
    override val y get() = state.y

    override val viewDistance: Int = 10
}
